pkg/trimaran/loadvariationriskbalancing/analysis.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package loadvariationriskbalancing
+
+import (
+	"math"
+
+	"github.com/paypal/load-watcher/pkg/watcher"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+)
+
+/*
+Calculation of risk score for resources given measured data
+*/
+
+// resourceStats : statistics data for a resource
+type resourceStats struct {
+	// average used (absolute)
+	usedAvg float64
+	// standard deviation used (absolute)
+	usedStdev float64
+	// req of pod
+	req float64
+	// node capacity
+	capacity float64
+}
+
+// computeScore : compute score given usage statistics
+/* risk = [ average + margin * stDev^{1/sensitivity} ] / 2 */
+/* score = ( 1 - risk ) * maxScore */
+func (rs *resourceStats) computeScore(margin float64, sensitivity float64) float64 {
+	if rs.capacity <= 0 {
+		klog.Errorf("invalid resource capacity %f!", rs.capacity)
+		return 0
+	}
+	rs.withinBounds()
```

Comment:
should we move this to L49 as capacity may be bounded to 0 in `withinBounds()`?

Reply:
Actually condition in L49 causes function to return 0, otherwise we'll have divide by 0 or undesirable negative quantities. Thus, no need to bound capacity in withinBounds(). I will remove it.